frontend/src/composables/use-sensitive-media.ts

```diff
@@ -0,0 +1,83 @@
+import { computed, watch } from "vue"
+
+import type { Media } from "~/types/media"
+
+import { useUiStore } from "~/stores/ui"
+
+type SensitiveVisibilityState =
+  | "insensitive"
+  | "sensitive-shown"
+  | "sensitive-hidden"
+
+/**
+ * A helper composable for working with sensitive media.
+ * Provides the current visibility of a sensitive media,
+ * along with toggles for hiding/revealing the media.
+ */
+export function useSensitiveMedia(
+  media: Pick<Media, "id" | "isSensitive"> | null
+) {
+  const uiStore = useUiStore()
+
+  // Clear out previously-revealed results when the toggle is switched.
+  watch(
+    () => uiStore.shouldBlurSensitive,
+    () => {
+      uiStore.revealedSensitiveResults = []
+    },
+    { deep: true }
+  )
+
+  /**
+   * The current state of a single sentive media item.
+   * Respects defaults from the site's filters and is
+   * updated by user interactions.
+   */
+  const visibility = computed<SensitiveVisibilityState>(() => {
+    if (!media) {
+      return "insensitive"
+    } else if (media.isSensitive) {
+      if (uiStore.revealedSensitiveResults.includes(media.id)) {
+        return "sensitive-shown"
+      } else {
+        return uiStore.shouldBlurSensitive
+          ? "sensitive-hidden"
+          : "sensitive-shown"
+      }
+    } else {
+      return "insensitive"
+    }
+  })
+
+  function reveal() {
+    if (media && !uiStore.revealedSensitiveResults.includes(media.id)) {
+      uiStore.revealedSensitiveResults.push(media.id)
+    }
+  }
+
+  function hide() {
+    if (!media) return
+    const index = uiStore.revealedSensitiveResults.indexOf(media.id)
```

Comment:
If we use a `set` for `revealedSensitiveResults`, we won't have to check whether the item is already in the array or not. 
Pinia didn't support `Set`s and `Map`s when hydrating on SSR, but the support was added in https://github.com/vuejs/pinia/issues/1608 (not that we really need SSR support since the reveal can only happen client side, right?)

Reply:
I believe array performance is still a bit better than Set performance in browsers, so I don't have a strong preference here. if you think it's a blocker I'm fine to update it though!

Reply:
First of all, let me say that it's not a blocker :) I don't think there would be a huge difference in performance, especially since we would probably have no more than 100 elements in the `revelaedSensitiveResults` anyways. 

I got curious about the browser performance of Sets, and it _seems_ that it's OK (and much faster for some operations with > 10000 items, which is not our case though)